cypress/support/step_definitions/getterSetterCalls.js

```diff
@@ -16,25 +16,509 @@
+      }
+
+      cy.then(() => {
+        let method =
+          methodType === CONSTANTS.SET
+            ? UTILS.resolveRecursiveValues(fireboltCallObject.setMethod)
+            : UTILS.resolveRecursiveValues(fireboltCallObject.method);
+
+        let validationJsonPath =
+          methodType === CONSTANTS.SET
+            ? UTILS.resolveRecursiveValues(fireboltCallObject.setValidationJsonPath)
+            : UTILS.resolveRecursiveValues(fireboltCallObject.validationJsonPath);
+
+        let contentObject =
+          methodType === CONSTANTS.SET
+            ? UTILS.resolveRecursiveValues(fireboltCallObject.setContent)
+            : UTILS.resolveRecursiveValues(fireboltCallObject.content);
+
+        method = method.includes('_') ? method.split('_')[1] : method;
+        contentObject = contentObject ? contentObject : CONSTANTS.NULL_RESPONSE;
+        validationJsonPath = validationJsonPath ? validationJsonPath : CONSTANTS.RESULT;
+
+        if (expectingError) {
+          contentObject = UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[
+            errorContent
+          ];
+        }
+        const additionalParams = {
+          method: method,
+          context: context,
+          validationJsonPath: validationJsonPath,
+          contentObject: contentObject,
+          expectingError: expectingError,
+          appId: appId,
+        };
+        if (!Cypress.env(CONSTANTS.SKIPCONTENTVALIDATION)) {
+          cy.methodOrEventResponseValidation(CONSTANTS.METHOD, additionalParams);
+        } else {
+          cy.log(
+            `${CONSTANTS.SKIPCONTENTVALIDATION} flag is enabled, Skipping the Content validation`
+          );
+        }
+      });
+    } else {
+      assert(false, `${sdk} SDK not Supported`);
+    }
+  }
+);
+
+/**
+ * @module getterSetterCalls
+ * @function Given '(.+)' platform (triggers|does not trigger) '(.*?)'(?: '(.*?)')? event(?: with '(.+)')?
+ * @description Performing a event validation against the source of truth
+ * @param {String} sdk - name of the sdk.
+ * @param {String} eventExpected - Determines whether the event is expected or not.
+ * @param {String} appId - app identifier.
+ * @param {String} fireboltCallKey - key name passed to look for firebolt call object in fireboltCallData Json.
+ * @param {String} errorContent - Doing error content validation when error content object key passed. Ex: 'INVALID_TYPE_PARAMS'
+ * @example
+ * And 'Firebolt' platform triggers '1st party app' 'CLOSEDCAPTION_SETTINGS' event
+ * And 'Firebolt' platform triggers '1st party app' event
+ * And 'Firebolt' platform triggers '3rd party app' 'CLOSEDCAPTION_SETTINGS' event
+ * And 'Firebolt' platform does not trigger '3rd party app' 'CLOSEDCAPTION_SETTINGS' event
+ * And 'Firebolt' platform triggers '1st party app' event
+ * And 'Firebolt' platform triggers '1st party app' 'CLOSEDCAPTION_SETTINGS' event with 'INVALID_TYPE_PARAMS'
+ */
+Given(
```

Comment:
Wondering if this one and the one above could also use the a single helper function?

Reply:
In upcoming enhancement ticket observed we are going to modify the glue and removing some fields, After removing some of the lines we need to remove. Is it to handle thins in new ticket?